c2rust-analyze/src/rewrite/expr/convert.rs

```diff
@@ -566,7 +679,7 @@ fn generate_zeroize_code(zero_ty: &ZeroizeType, lv: &str) -> String {
     match *zero_ty {
         ZeroizeType::Int => format!("{lv} = 0"),
         ZeroizeType::Bool => format!("{lv} = false"),
-        ZeroizeType::Iterable(ref elem_zero_ty) => format!(
```

Comment:
Why the change here from `Iterable` to `Array`? I'm assuming `Iterable` is a superset of the other, so what gets left out?

Reply:
Answered my own question: it's just a rename.

Reply:
It was previously called `Iterable` because it could in principle be used for anything with an `iter_mut()` method.  It's now called `Array` because it can only be used for arrays specifically - the `generate_zeroize_expr` function used for initializing allocations can't be generic as the code we use for zeroizing an existing value on `memset(_, 0, _)`.